mso/resource_mso_schema_template_contract.go

```diff
@@ -48,511 +50,515 @@ func resourceMSOTemplateContract() *schema.Resource {
+				contractDetails := contract.(map[string]interface{})
+				if contractDetails["name"].(string) == contractName {
+					d.SetId(fmt.Sprintf("/templates/%s/contracts/%s", templateName, contractName))
 					d.Set("schema_id", schemaId)
-					d.Set("template_name", apiTemplate)
-					d.Set("display_name", models.StripQuotes(contractCont.S("displayName").String()))
-					d.Set("filter_type", models.StripQuotes(contractCont.S("filterType").String()))
-					d.Set("scope", models.StripQuotes(contractCont.S("scope").String()))
-
-					count, _ := contractCont.ArrayCount("filterRelationships")
-
-					var filterSchema string
-					var filterTemplate string
-					var filterName string
+					d.Set("template_name", templateName)
+					d.Set("contract_name", contractName)
 
-					if tempVar, ok := d.GetOk("filter_relationships"); ok {
-						filter_relationships := tempVar.(map[string]interface{})
+					if val, ok := contractDetails["displayName"]; val != nil && ok {
+						d.Set("display_name", val.(string))
+					}
+					if val, ok := contractDetails["scope"]; val != nil && ok {
+						d.Set("scope", val.(string))
+					}
+					if val, ok := contractDetails["filterType"]; val != nil && ok {
+						d.Set("filter_type", val.(string))
+					}
+					if val, ok := contractDetails["prio"]; val != nil && ok {
+						d.Set("priority", val.(string))
+					}
+					if val, ok := contractDetails["targetDscp"]; val != nil && ok {
+						d.Set("target_dscp", val.(string))
+					}
 
-						if filter_relationships["filter_schema_id"] != nil {
-							filterSchema = filter_relationships["filter_schema_id"].(string)
+					filterList := []map[string]interface{}{}
+					if val, ok := contractDetails["filterRelationships"]; val != nil && ok {
+						filterList = setFilterRelationshipList(val.([]interface{}), filterList, "bothWay")
+
+						// TODO Remove below block of code once the filterRelationships is deprecated
+						// Start of block
+						// Reason for adding this logic is backworth compatibility with previous release
+
+						if val, ok := d.GetOk("filter_relationships"); val != nil && ok {
+							filterRelationshipsMap := val.(map[string]interface{})
+							filterSchemaId := schemaId
+							if filterRelationshipsMap["filter_schema_id"] != nil {
+								filterSchemaId = filterRelationshipsMap["filter_schema_id"].(string)
+							}
+							filterTemplateName := templateName
+							if filterRelationshipsMap["filter_template_name"] != nil {
+								filterTemplateName = filterRelationshipsMap["filter_template_name"].(string)
+							}
+							filterName := filterRelationshipsMap["filter_name"].(string)
+
+							for _, fiterMap := range filterList {
+								if fiterMap["filter_schema_id"].(string) == filterSchemaId &&
+									fiterMap["filter_template_name"].(string) == filterTemplateName &&
+									fiterMap["filter_name"].(string) == filterName {
+									d.Set("filter_relationships", filterRelationshipsMap)
```

Comment:
Just for a note, it should be TypeSet or List of Map, as per the current code it stores only the last object information in the state file.

Reply:
correct this is the intention, this is the old format which needed to be set back to original value